### ISSUE TITLE
Handle WSL and SSH sessions in `ls` clickable links

### ISSUE THREAD

**Issue by rgwood** [nushell/nushell#6436]
### Related problem

[The new clickable links](https://github.com/nushell/nushell/pull/6333) are extremely cool. But they (understandably) don't work in WSL or in remote SSH sessions.

### Describe the solution you'd like

I think we should:

- Detect when we're running in WSL and translate paths using [`wslpath`](https://devblogs.microsoft.com/commandline/windows10v1803/#interoperability)
    - ex: `/home/reilly` -> `\\wsl.localhost\Ubuntu\home\reilly`
- Detect when we're in a remote SSH session and disable the links

### Additional context and details

I'd be happy to implement this if there are no objections.

**Comment by fdncred** [comment]
I haven't studied the wsl-rs crate in detail but I do know that wsl has environment variables that are unique to it. So, it could be as simple as checking for one of those. I'm no my Mac but it seems like there's 2 or 3 env vars. I know vscode won't work right without one of them.

**Comment by rgwood** [comment]
Good idea; checking for env vars (`WSLENV`, `WSL_DISTRO_NAME`, `WSL_INTEROP`) is probably easier than adding yet another dependency. I'm also not optimistic about my PR to `wsl-rs` being accepted, seems like that crate might be abandoned.

**Comment by rgwood** [comment]
Tried implementing proper WSL support. Unfortunately, it looks like Windows Terminal doesn't support UNC paths:
![image](https://user-images.githubusercontent.com/26268125/187089208-9894fac4-2e6e-49c0-9df9-a92643fd0956.png)

Will investigate later, but maybe for now we just disable the links when running in WSL or SSH.

**Comment by fdncred** [comment]
I was thinking there was a `\\wsl$` special UNC path or a wslpath ... I'd have to google. I wonder what Gnome terminal and other native linux terminals do, or even wslg terminals. I have wslg installed and can check tomorrow if I remember.

**Comment by rgwood** [comment]
> I was thinking there was a \\wsl$ special UNC path or a wslpath

My understanding is that the official way to get a Windows path is `wslpath -w /foo` (which returns `\\wsl.localhost\Ubuntu\foo` or similar). I tried that but it was a no go.

**Comment by fdncred** [comment]
After trying a few ideas and inspecting the WT code, I think you're correct.

The message we see in our popups with `file://` is from here `UnsupportedSchemeText`:
```cpp
    void TerminalPage::_OpenHyperlinkHandler(const IInspectable /*sender*/, const Microsoft::Terminal::Control::OpenHyperlinkEventArgs eventArgs)
    {
        try
        {
            auto parsed = winrt::Windows::Foundation::Uri(eventArgs.Uri().c_str());
            if (_IsUriSupported(parsed))
            {
                ShellExecute(nullptr, L"open", eventArgs.Uri().c_str(), nullptr, nullptr, SW_SHOWNORMAL);
            }
            else
            {
                _ShowCouldNotOpenDialog(RS_(L"UnsupportedSchemeText"), eventArgs.Uri());
            }
        }
        catch (...)
        {
            LOG_CAUGHT_EXCEPTION();
            _ShowCouldNotOpenDialog(RS_(L"InvalidUriText"), eventArgs.Uri());
        }
    }
```
Which calls `_IsUriSupported` here:

```cpp
    // Method Description:
    // - Determines if the given URI is currently supported
    // Arguments:
    // - The parsed URI
    // Return value:
    // - True if we support it, false otherwise
    bool TerminalPage::_IsUriSupported(const winrt::Windows::Foundation::Uri& parsedUri)
    {
        if (parsedUri.SchemeName() == L"http" || parsedUri.SchemeName() == L"https")
        {
            return true;
        }
        if (parsedUri.SchemeName() == L"file")
        {
            const auto host = parsedUri.Host();
            // If no hostname was provided or if the hostname was "localhost", Host() will return an empty string
            // and we allow it
            if (host == L"")
            {
                return true;
            }
            // TODO: by the OSC 8 spec, if a hostname (other than localhost) is provided, we _should_ be
            // comparing that value against what is returned by GetComputerNameExW and making sure they match.
            // However, ShellExecute does not seem to be happy with file URIs of the form
            //          file://{hostname}/path/to/file.ext
            // and so while we could do the hostname matching, we do not know how to actually open the URI
            // if its given in that form. So for now we ignore all hostnames other than localhost
        }
        return false;
    }
```
so you can see that if it's not `localhost` then it's not going to work at all.

I did have success with `/mnt/c/Windows/explore.exe \\wsl$\Ubuntu\home\fdncred` but there's no real way to make that work here.

**Comment by NotTheDr01ds** [comment]
> > I was thinking there was a \wsl$ special UNC path or a wslpath
> 
> My understanding is that the official way to get a Windows path is `wslpath -w /foo` (which returns `\\wsl.localhost\Ubuntu\foo` or similar). I tried that but it was a no go.

Not that it matters, but `\\wsl.localhost\<distro>` is Windows 11-specific.  Windows 10 still uses `\\wsl$\<distro>`.  `wslpath -w` should return the correct form regardless.

>  it looks like Windows Terminal doesn't support UNC paths

I'll try to update, look at how this is implemented, and noodle on it.  Might still be some way around it, but yeah, if we have to rely on Windows Terminal interpreting the links, then we might be out of luck.  It looks like there's already a [related issue](https://github.com/microsoft/terminal/issues/10188) open for WT, but it also looks like they need some help from the Windows team to move forward.

**Comment by rgwood** [comment]
Good find and write up, thanks
